src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/CallInstruction.cs

```diff
@@ -122,6 +122,9 @@ public static CallInstruction Create(MethodInfo info, ParameterInfo[] parameters
 #endif
         }
 
+        [DynamicDependency("GetValue", typeof(Array))]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2075:UnrecognizedReflectionPattern",
+            Justification = "The GetValue methods will be preserved by the DynamicDependency.")]
```

Comment:
@MichalStrehovsky  will confirm, but I would expect the `GetValue` to be a runtime generated method, just like the array .ctors (because it has different overloads based on the array rank).

Reply:
I didn't think it was generated by looking at the code. It looks like the first 3 are in IL.

https://github.com/dotnet/runtime/blob/8a52f1e948b6f22f418817ec1068f07b8dae2aa5/src/coreclr/System.Private.CoreLib/src/System/Array.CoreCLR.cs#L320-L357

And in ILSpy:

![image](https://user-images.githubusercontent.com/8291187/106201598-4bc66980-617e-11eb-96c3-e3d955777903.png)

Update: They can definitely get trimmed, I tested in a Blazor WASM app on 5.0:

```C#
        int[,,] a = new int[3, 2, 1];
        Console.WriteLine(a.GetType());
        Console.WriteLine(a.GetType().Name);
        Console.WriteLine(a.GetType().Assembly);
        foreach (var m in a.GetType().GetMethods())
        {
            Console.WriteLine(m.Name);

            if (m.Name == "GetValue")
            {
                Console.WriteLine($"\t{m.GetParameters().Length}");
            }
        }
```

Not published prints:
![image](https://user-images.githubusercontent.com/8291187/106202013-ef177e80-617e-11eb-8857-34ffd753aabd.png)

publishing and trimming the app prints:

![image](https://user-images.githubusercontent.com/8291187/106202042-fb9bd700-617e-11eb-952a-52d1719b262b.png)

Reply:
Wow - thanks for the learning experience :smile:

Reply:
The runtime generated methods are named "Get", "Set" and "Address". Seems like this is doing a translation from that, for whatever reason (see the `bool isGetter = info.Name == "Get";` line all the way up).